src/hotspot/share/cds/heapShared.hpp

```diff
@@ -253,6 +253,13 @@ class HeapShared: AllStatic {
       HeapShared::oop_hash> ArchivedObjectCache;
   static ArchivedObjectCache* _archived_object_cache;
 
+  typedef ResourceHashtable<oop, oop,
+      15889, // prime number
+      ResourceObj::C_HEAP,
+      mtClassShared,
+      HeapShared::oop_hash> OriginalObjectTable;
+  static OriginalObjectTable* _original_object_table;
```

Comment:
Did a small test, and a hello world creates a map with 33k objects. So this will have a load factor > 2, is that okay? 

Maybe we should create and fill this map only when logging. Its only used for logging, right? Why pay for this map always. Archive creation at build is already kind of slow.

Reply:
I changed the table size to 36137 and also use this table only when `-Xlog:cds+map` is enabled.